src/audio/data_blob.c

```diff
@@ -272,6 +272,134 @@ int comp_data_blob_set(struct comp_data_blob_handler *blob_handler,
 	return 0;
 }
 
+int ipc4_comp_data_blob_set(struct comp_data_blob_handler *blob_handler,
```

Comment:
We already have a function performs the same blob set behavior, see `comp_data_blob_set` in data_blob.c, I think you can reuse `comp_data_blob_set`

Reply:
@ujfalusi @jsarha - fyi can you check the driver is sending the fragments in the correct order using the 0x2 and 0x1 above as described by @iganakov . Thanks

Reply:
@ujfalusi @jsarha ping - is driver sending correct bits in correct order ?

Reply:
Indeed this looks to me that the fragments are sent in sequence starting from zero offset:
https://github.com/thesofproject/linux/blob/ddc2987f8e32125e6e783e582e44babc303a70b8/sound/soc/sof/ipc4.c#L429

So how should the fragments be sent? Starting from the end, offset running down to zero? Should the highest offset have first block bit (extension bit 29) set, and the low offset - that is sent last - the last block (bit 28) set?

And corresponding change also needed for SOF FW? (ATM It appears to expect the fragments to begin from low offset to high)

Reply:
@jsarha driver aligned ?

Reply:
Great - we are aligned.